ts/client/src/accounts/healthCache.ts

```diff
@@ -1159,6 +1160,52 @@ export class HealthCache {
 
     return baseLots.floor();
   }
+
+  public getPerpPositionLiquidationPrice(
+    group: Group,
+    mangoAccount: MangoAccount,
+    perpPosition: PerpPosition,
+  ): I80F48 {
+    const hc = HealthCache.fromMangoAccount(group, mangoAccount);
+    const perpMarket = group.getPerpMarketByMarketIndex(
+      perpPosition.marketIndex,
+    );
+
+    function healthAfterPriceChange(newPrice: I80F48): I80F48 {
+      const gClone = cloneDeep(group);
+      gClone.getPerpMarketByMarketIndex(perpMarket.perpMarketIndex)._price =
+        newPrice;
+      const hc = HealthCache.fromMangoAccount(gClone, mangoAccount);
```

Comment:
we could save a bunch of compute here by not rebuilding the HealthCache from scratch and just overriding the perp price. But what you're doing now is true black-box use of HealthCache, so maybe most stable